assets/src/dashboard/animations/effects/whooshIn/index.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { AnimationMove } from '../../parts/move';
+import { AnimationFade } from '../../parts/fade';
+import { AnimationZoom } from '../../parts/zoom';
+import getOffPageOffset from '../../utils/getOffPageOffset';
+import { DIRECTION } from '../../constants';
+
+export function EffectWhooshIn({
+  duration = 400,
+  whooshInDir = DIRECTION.LEFT_TO_RIGHT,
+  easing = 'cubic-bezier(0.0, 0.0, 0.2, 1)',
+  delay,
+  element,
+}) {
+  const id = uuidv4();
+
+  const { offsetLeft, offsetRight } = getOffPageOffset(element);
+
+  const offsetX =
+    whooshInDir === DIRECTION.RIGHT_TO_LEFT
+      ? `${offsetRight}%`
+      : `${offsetLeft}%`;
+
+  const {
+    WAAPIAnimation: moveWAAPIAnimation,
+    AMPTarget: moveAMPTarget,
+    AMPAnimation: moveAMPAnimation,
+    generatedKeyframes: moveKeyframes,
+  } = AnimationMove({
+    offsetX,
+    duration,
+    delay,
+    easing,
+  });
+
+  const {
+    WAAPIAnimation: fadeWAAPIAnimation,
+    AMPTarget: fadeAMPTarget,
+    AMPAnimation: fadeAMPAnimation,
+    generatedKeyframes: fadeKeyframes,
+  } = AnimationFade({
+    fadeFrom: 0,
+    fadeTo: 1,
+    duration,
+    delay,
+    easing,
+  });
+
+  const {
+    WAAPIAnimation: zoomWAAPIAnimation,
+    AMPTarget: zoomAMPTarget,
+    AMPAnimation: zoomAMPAnimation,
+    generatedKeyframes: zoomKeyframes,
+  } = AnimationZoom({
+    zoomFrom: 0.15,
+    zoomTo: 1,
+    duration,
+    delay,
+    easing,
+  });
+
+  return {
+    id,
+    WAAPIAnimation: ({ children, ...args }) =>
+      moveWAAPIAnimation({
+        children: fadeWAAPIAnimation({
+          children: zoomWAAPIAnimation({ children, ...args }),
+          ...args,
+        }),
+        ...args,
+      }),
```

Comment:
is there any benefit of using the functional syntax here over jsx?
```
WAAPIAnimation: ({ children, hoistAnimation }) => (
  <MoveWAAPIAnimation hoistAnimation={hoistAnimation}>
    <FadeWAAPIAnimation hoistAnimation={hoistAnimation}>
      <ZoomWAAPIAnimation hoistAnimation={hoistAnimation}>
        {children}
      </ZoomWAAPIAnimation>
    </FadeWAAPIAnimation>
  </MoveWAAPIAnimation>
),
```

If we find ourselves doing this a lot too, can alway export the more terse `<ComposableWrapper />` from `storyAnimation/WAAPIWrapper.js` to accomplish the same thing:

```
<ComposableWrapper 
  /* could always update this to just take array of WAPPIAnimations too */
  animationParts={[
    {WAAPIAnimation: moveWAAPIAnimation},
    {WAAPIAnimation: fadeWAAPIAnimation},
    {WAAPIAnimation: zoomWAAPIAnimation}
  ]}
  hoistAnimation={hoistAnimation}
>
  {children}
</ComposableWrapper>
```

Reply:
Ooo, I didn't even think about using the jsx syntax, I'll go head and re-write that way, I think it's much more clear what's happening that way.  

And I agree with your composable wrapper concept too if we find our selves doing this a bunch.